pkg/storage/ingest/pusher.go

```diff
@@ -66,7 +66,7 @@ func (c pusherConsumer) Consume(ctx context.Context, records []record) (returnEr
 	}(time.Now())
 
 	type parsedRecord struct {
-		*mimirpb.WriteRequest
+		*mimirpb.PreallocWriteRequest
```

Comment:
This PR depends on https://github.com/grafana/mimir/pull/11393. I inlined those changes into this PR until that's merged.

This is logically equivalent to:
```
WriteRequest: &mimirpb.WriteRequest{
	Timeseries: mimirpb.PreallocTimeseriesSliceFromPool(),
},
```
PreallocWriteRequest.Unmarshal() calls PreallocTimeseriesSliceFromPool()